### PR TITLE
refactor: rewrite `EntryPoint` to use `SoftFailure`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/5500

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Validation itself contains a list, so there is no need for an extra list.

**Review by mlutze — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala [hunk 1]
```diff
@@ -134,24 +134,22 @@ object EntryPoint {
...
+        case _ :: _ :: _ => None.toSoftFailure(EntryPointError.IllegalEntryPointArgs(sym, sym.loc))
         // Case 3: Empty arguments. Impossible since this is desugared to Unit.
         case Nil => throw InternalCompilerException("Unexpected empty argument list.", loc)
       }
 
-      flatMapN(argVal: Validation[Type, EntryPointError]) {
-        arg =>
-          val argSc = Scheme.generalize(Nil, Nil, arg)
-
-          if (Scheme.equal(unitSc, argSc, classEnv, ListMap.empty)) { // TODO ASSOC-TYPES better eqEnv
```
[the remark this answers is not in the document]
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:148]: Yeah sorry about that! 😄

**Comment by magnus-madsen** [comment]
Status?

**Comment by jaschdoc** [comment]
Ready to be merged :)

File: main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala [hunk 2]
```diff
@@ -134,24 +134,23 @@ object EntryPoint {
...
-          }
+      flatMapN(argVal: Validation[Option[Type], EntryPointError]) {
+        // Case 1: Unit -> XYZ. We can ignore the args.
+        case Some(arg) if Scheme.equal(unitSc, Scheme.generalize(Nil, Nil, arg), classEnv, ListMap.empty) =>
+          // TODO ASSOC-TYPES better eqEnv
+          ().toSuccess
+
+        // Case 2: Bad arguments. Error
+        // Case 3: argVal was None / SoftError
+        case Some(_) | None =>
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:152]: This code is really ugly. Can we not do better?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:152]: :)

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:152]: Oh oh... what style do you prefer / is better?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:152]: I just find the whole pattern match so confusing. Like when does whole succeed or fail?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:152]: @mlutze

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala]: Better with a wildcard?

**Pull request merged**